doc/source/serve/http-guide.md

```diff
@@ -251,3 +190,11 @@ Here is a list of adapters; please feel free to [contribute more](https://github
 
 ```
 
+## Choosing the right HTTP feature
+
+Serve offers a layered approach to expose your model with the right HTTP API.
+
+Considering your use case, you can choose the right level of abstraction:
+- If the built-in adapters fit your use case or you can supply your own adapter, use `DAGDriver` with `http_adapter`.
+- If you are comfortable working with the raw request object, use `starlette.request.Requests` API.
+- If you want a fully fledge API server with validation and doc generation, use the FastAPI integration.
```

Comment:
This is really valuable information, maybe we should put it at the top / in the intro to the section with links to the subsections?